### ISSUE TITLE
Attachments arrive unnamed

### ISSUE THREAD

**Issue by gigapepe2000** [deltachat/deltachat-android#1623]
Hi.
When I attach a file (doc, xls, pdf), the recipient (eg gmail) receives it without a name and without an extension ("noname"). What should I do?
Thank you

[Hola.
Cuando adjunto un archivo (doc, xls, pdf), el destinatario (por ejemplo gmail) lo recibe sin nombre y sin extension ("noname"). Que debo hacer?
Gracias]

- Operating System (Android): 6.0.1
- Device: LG
- Delta Chat Version: 1.12.5
- Expected behavior: send attachments
- Actual behavior: attachments arrive unnamed
- Steps to reproduce the problem: send attachment to gmail address

**Comment by gigapepe2000** [comment]
.txt and .html files reach the recipient fine

[archivos .txt y .html llegan bien al destinatario]

**Comment by r10s** [comment]
i just had a look, this is what delta produced as an email with an pdf attached: https://gist.github.com/r10s/d0b80dd915df6ebdee65f86f604dd708 - at a first glance this looks good to me.

i tested if this renders correctly in thunder bird as well as in gmail, also that looks good to me:

![Screen Shot 2020-09-15 at 15 24 36](https://user-images.githubusercontent.com/9800740/93216462-0d4c8700-f768-11ea-9cea-e192aa1968c2.png)

can you please retry and/or add additional information so that we have a chance to reproduce the issue?

does anyone else have a similar issue?

**Comment by gigapepe2000** [comment]
![Captura3](https://user-images.githubusercontent.com/56970801/94434523-39213100-019a-11eb-9d9a-d6825300923c.PNG)


recently they arrive with the correct icon but no name

**Comment by Hocuri** [comment]
How did you choose the attatchments? Click share and then select Delta Chat? Or select "File" from the attatchments menu in Delta Chat?

Also how do they look in Delta Chat? Do they have the right name there?

**Comment by gigapepe2000** [comment]
I have tried both ways. (from the file, and from the attatchments menu in Delta Chat)
They look good in delta, just add "-1" at the end of the name. 

I tried now to send an excel (.xlsx) and the recipient (gmail) receives it without name but with the correct icon. 
Then I send a excel (.xls) and it lost name and icon.

**Comment by agutierrez25249** [comment]
@r10s 

I just checked the error. It happens when you send a document with spaces in the name, I tried with * .txt, * .xlsx, in both cases «Gmail-web» showed them as «noname»

Delta v1.13.4

**Comment by Hocuri** [comment]
Ok I could reproduce this now. GMail just uses a different encoding for special characters (which a space is).
Sending two files called `t" t.txt`:
GMail:
```
Content-Type: text/plain; charset="US-ASCII"; name="t\" t.txt"
Content-Disposition: attachment; filename="t\" t.txt"
Content-Transfer-Encoding: base64
Content-ID: <f_kfqxhx290>
X-Attachment-Id: f_kfqxhx290
```
Delta Chat:
```
Content-Type: application/octet-stream
Content-Disposition: attachment; filename*="=?utf-8?q?t=22_t=2Etxt?="
Content-Transfer-Encoding: base64
```
DC does not parse GMail's encoding perfectly, either: My test file GMail->DC arrived as <code>&nbsp;t.txt</code>.

No idea what to do about this.

**Comment by gigapepe2000** [comment]
Take a look at this, I think it could be useful

I sent: prueba.rar		
gmail received: prueba-1.rar
infomed received: prueba-1.rar

I sent: second test.rar		
gmail received: noname
infomed received: %?utf-8?q?second test-2.rar?%
			
I sent: acentuó.rar		
gmail received: noname
infomed received: %?utf-8?q?acentuó-2-1.rar?%

the error seems to be related to special characters in general		
but I don't know why delta adds -1, -2, ... to the end of the file names

**Comment by gigapepe2000** [comment]
Oh, thank you both very much.
Meanwhile, I will send the names of the files without special characters  :-)

**Comment by r10s** [comment]
with very good chances, this will be fixed with the upcoming android release; we fixed several issues wrt filenames, e.g.

https://github.com/deltachat/deltachat-core-rust/pull/2080  
https://github.com/deltachat/deltachat-core-rust/pull/2094  
https://github.com/deltachat/deltachat-core-rust/pull/2102

so, this needs to be retried with the next official release or with a nightly using current master.

**Comment by Hocuri** [comment]
@gigapepe2000 version 1.20 is released, can you try out if it works fine now?

**Comment by r10s** [comment]
closing as there is no actionable item left and many things are fixed; if there are still/new issues with 1.20.5, we should open a new issue.
